### DISCUSSION TITLE
Recreate Depth of Terrain

### DISCUSSION THREAD

**Discussion by dennemark** [maplibre/maplibre-gl-js#4892] [Q&A]
Hi,

I am rendering a 3d-layer on top of my map. Unfortunately at some point i have to clear the depth stencil. And this leads to an issue if i want to render maplibre layers on top of my 3d-layer. Precisely it is the seams that become visible between tiles. Since the terrain depth is missing, maplibre will render new layers with these seams.
Now I am wondering if it would be possible to either transfer the depth stencil from the layer before the 3d-renderer to the one after. Or probably much simpler: create a depth map by recreating a simple terrain layer. I tried to do the latter by just adding a background layer. It works if the background layer is opaque, but does not work if it is transparent.
```
map.addLayer({

        id: "depthTerrain"
        type: "background"
        beforeId: "3d"
        layout: { visibility: 'visible' } // 'none' does not work.
        paint: {
          'background-color': '#ffffff',
          'background-opacity': 1 // 0 does not work, 1 works but then i do not see anything below this map of course...
        }
      })
```

Someone has an idea?

Here we see the seam and also if i make the background layer transparent to 0.5. So it is also affected by missing depth.
![grafik](https://github.com/user-attachments/assets/ce0fdb7a-7b98-4e0b-b4fb-608b3e6781c0)

**Comment by HarelM** [comment]
Which version are you using? There was a change recently to help globe with custom layer by creating some depth buffer...?

**Reply by dennemark** [reply]
Hi,
Thanks for your reply! I am on version 4.5.2 but could upgrade to 4.7.1.
Is the upgrade already changing the behaviour? Or is there some way I can create the depth buffer?
Best

**Reply by HarelM** [reply]
Version 5.x has this, not 4, but it might not be what you need...

**Reply by dennemark** [reply]
Ah thanks!  I guess it is something like this: `painter.getDepthModeForSublayer(0, DepthMode.ReadOnly);`

Found here: https://github.com/maplibre/maplibre-gl-js/pull/4838/files#

Going to try it out.
